internal/api/identity.go

```diff
@@ -0,0 +1,217 @@
+package api
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	core "github.com/iden3/go-iden3-core/v2"
+	"github.com/iden3/go-iden3-core/v2/w3c"
+
+	"github.com/polygonid/sh-id-platform/internal/common"
+	"github.com/polygonid/sh-id-platform/internal/core/ports"
+	"github.com/polygonid/sh-id-platform/internal/core/services"
+	"github.com/polygonid/sh-id-platform/internal/gateways"
+	"github.com/polygonid/sh-id-platform/internal/kms"
+	"github.com/polygonid/sh-id-platform/internal/log"
+)
+
+// CreateIdentity is created identity controller
+func (s *Server) CreateIdentity(ctx context.Context, request CreateIdentityRequestObject) (CreateIdentityResponseObject, error) {
+	method := request.Body.DidMetadata.Method
+	blockchain := request.Body.DidMetadata.Blockchain
+	network := request.Body.DidMetadata.Network
+	keyType := request.Body.DidMetadata.Type
+
+	if keyType != "BJJ" && keyType != "ETH" {
+		return CreateIdentity400JSONResponse{
+			N400JSONResponse{
+				Message: "Type must be BJJ or ETH",
+			},
+		}, nil
+	}
+
+	rhsSettings, err := s.networkResolver.GetRhsSettingsForBlockchainAndNetwork(ctx, blockchain, network)
```

Comment:
I little bit confused with this part. Need some time to understand it.

Reply:
I was confused about the naming. The method `GetRhsSettingsForBlockchainAndNetwork` should be renamed to `GetRevocationSettingsForBlockchainAndNetwork` because this method also returns information for other revocation statuses, not just the RHS revocation status.

Secondly, I think I missed this earlier, but we should change this (not in this PR). Why do we define the revocation status not for the issuer but for the chain? On the old issuer server, we had different logic to manage different revocation statuses. On the old issuer server, it was possible to create an issuer and specify the type of revocation for the issuer:
POST /identitye/create
```json
{
  blockchain:
  method:
  network:
  type:
authBJJCredentialStatus: <where we can set one value from the list: Iden3ReverseSparseMerkleTreeProof|SparseMerkleTreeProof|...|etc"
}
```

Reply:
Good point. Let's not forget about this and find a time to discuss it. @martinsaporiti FYI.